lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class DomWindow {}
+
+extension DomWindowExtension on DomWindow {
+  external DomDocument get document;
+}
+
+@JS('window')
+external DomWindow get domWindow;
+
+@JS()
+@staticInterop
+class DomDocument {}
+
+extension DomDocumentExtension on DomDocument {
+  external List<DomNode> querySelectorAll(String selectors);
+}
+
+@JS()
+@staticInterop
+class DomEventTarget {}
+
+@JS()
+@staticInterop
+class DomNode extends DomEventTarget {}
+
+@JS()
+@staticInterop
+class DomHTMLElement extends DomNode {}
+
+@JS()
+@staticInterop
+class DomHTMLMetaElement {}
+
+extension DomHTMLMetaElementExtension on DomHTMLMetaElement {
+  external String get name;
+  external String get content;
+
+  external set name(String value);
+}
+
+Object domGetConstructor(String constructorName) =>
+    js_util.getProperty(domWindow, constructorName);
+
+extension DomHTMLListExtension on Iterable<DomNode> {
+  Iterable<T> whereHTMLType<T>(String constructorName) =>
+      where((DomNode element) =>
+              js_util.instanceof(element, domGetConstructor(constructorName)))
```

Comment:
I don't think you technically need this, right? Using the string directly should work iirc, but I suppose this is cleaner.

Reply:
I think that is `typeof`. When I tried passing a string to `instanceof` it complained that `Error: Right-hand side of 'instanceof' is not an object`.

Reply:
Ah, that must mean it interprets the input as a string instead of directly evaluating it e.g. `'Window'` instead of `Window`. This LGTM then.